src/drivers/intel/ssp/ssp.c

```diff
@@ -87,9 +96,12 @@ static void ssp_empty_rx_fifo(struct dai *dai)
 	while ((ssp_read(dai, SSSR) & SSSR_RNE) && retry--) {
 		entries = SSCR3_RFL_VAL(ssp_read(dai, SSCR3));
 		dai_dbg(dai, "ssp_empty_rx_fifo(), before flushing, entries %d", entries);
-		for (i = 0; i < entries + 1; i++)
-			/* read to try empty fifo */
-			ssp_read(dai, SSDR);
+
+		if (read_ssdr) {
+			for (i = 0; i < entries + 1; i++)
+				/* read to try empty fifo */
+				ssp_read(dai, SSDR);
```

Comment:
nitpick: I would move the comment up, before the for() loop and reword it to `Empty the FIFO by reading out the entries`

Reply:
Done